### ISSUE TITLE
run_shell_script in deployment.steps.utility does not handle spaces properly on Windows devices

### ISSUE THREAD

**Issue by conewman1** [PrefectHQ/prefect#10705]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

In our prefect.yaml file we are executing commands using the step type of prefect.deployments.steps.run_shell_script.  If we include an argument that has a space in it, we enclose that in double quotes like you normally would on the command line.

The issue I am seeing is that in this line of code:
[https://github.com/PrefectHQ/prefect/blob/main/src/prefect/deployments/steps/utility.py#L172](https://github.com/PrefectHQ/prefect/blob/main/src/prefect/deployments/steps/utility.py#L172)

It executes **shlex.split(command)** which removes the double quotes.  Apparently there is a known issue where the double quotes need to be added back when on the Windows platform if the arguments are going to be passed back to the subprocess module.  Here is a relevant link with someone else having this issue with shlex:
[https://github.com/jrnl-org/jrnl/issues/348](https://github.com/jrnl-org/jrnl/issues/348)

The recommended workaround is to include the posix argument and set it to True or False based on the platform.

### Describe the proposed behavior

Update [this line](https://github.com/PrefectHQ/prefect/blob/main/src/prefect/deployments/steps/utility.py#L172) so that it looks like this:

`split_command = shlex.split(command, posix="win" not in sys.platform)`

Please note that this was recommended as a solution in the link provided under Current Behavior.   I tested this on a Windows 10 laptop and afterwards I was able to execute this deployment step successfully.  

### Additional context

While our deployed code runs on Linux containers, our developers do initial development and then run deployments to our Dev workspaces from Windows machines.  Not being able to run these utility steps from Windows machines is causing issues on the team.

**Comment by serinamarie** [comment]
Hi @conewman1, thanks so much for this issue! We'd welcome a contribution if you're interested :)

**Comment by NikoRaisanen** [comment]
I'd be happy to tackle this as well.

I want to give you the opportunity first if you want @conewman1 since you already found a solution 😄

**Comment by conewman1** [comment]
@NikoRaisanen please make this change if you don't mind.  I'm still getting used to Github and would prefer this be fixed sooner rather than later.   Thank you!

**Comment by NikoRaisanen** [comment]
Sounds great! I'll get working on it

**Comment by serinamarie** [comment]
Thanks @NikoRaisanen, I will assign this to you :)

**Comment by NikoRaisanen** [comment]
I opened up a draft PR so all of use here can look at the solution. I still need to set up Prefect locally and write the tests, just looking for some feedback to make sure I'm on the right path.

I decided to go with an explicit check for Windows with:  `split_command = shlex.split(command, posix=sys.platform != "win32")` because checking for just the substring "win" would also set `posix=False` if sys.platform is "darwin" (macOS)

[Here](https://docs.python.org/3/library/sys.html#:~:text=the%20values%20are%3A-,System,-platform%20value) is the official documentation on sys.platform return values. My understanding is that cygwin is also POSIX compliant, so it shouldn't need the posix=False argument -- but I'm not a cygwin expert by any means 

Curious to hear if anybody has thoughts / feedback
